### ISSUE TITLE
ERROR: TypeError: can't convert nil into String

### ISSUE THREAD

**Issue by ringods** [websterclay/knife-github-cookbooks#4]
I tried integrating the jenkins cookbook into my repo using your plugin, but it failed with the output as can be seen here:

rdesmet@ringo-desktop:~/infrastructure/chef-repo$ knife cookbook github install fnichol/chef-jenkins -V
Installing jenkins from git://github.com/fnichol/chef-jenkins.git to /home/rdesmet/infrastructure/chef-repo/.chef/../cookbooks
Checking out the master branch.
Pristine copy branch (chef-vendor-jenkins) exists, switching to it.
ERROR: TypeError: can't convert nil into String
rdesmet@ringo-desktop:~/infrastructure/chef-repo$ 

Tried this with 0.1.3 and 0.1.4. My knife.rb is on log_level :debug and I also tried with the -V on the command line, but I don't get more output than this.

Ringo

(BTW, where is the code commit with version.rb containing 0.1.4?)

**Comment by jnewland** [comment]
Whoops, I had forgotten to push my 'bump version' commit.

```
knife cookbook github install fnichol/chef-jenkins -V
Installing jenkins from git://github.com/fnichol/chef-jenkins.git to /Volumes/src/cookbooks/dev/.chef/../cookbooks
Checking out the master branch.
Creating pristine copy branch chef-vendor-jenkins
Removing pre-existing version.
1 files updated, committing changes
Creating tag cookbook-site-imported-jenkins-3eec6ac014156ce6f02d3ccb4bb42d3ea49ef1d3
Checking out the master branch.
Updating 50bf6b4..b54af0e
Fast-forward
 cookbooks/jenkins/.gitignore                       |    2 +
 cookbooks/jenkins/README.rdoc                      |  188 +++++++++++++++
 cookbooks/jenkins/attributes/default.rb            |  122 ++++++++++
 cookbooks/jenkins/files/default/node_info.groovy   |   86 +++++++
 cookbooks/jenkins/files/ubuntu/jenkins.list        |    1 +
 cookbooks/jenkins/libraries/manage_node.rb         |  158 +++++++++++++
 cookbooks/jenkins/metadata.rb                      |    7 +
 cookbooks/jenkins/providers/cli.rb                 |   54 +++++
 cookbooks/jenkins/providers/execute.rb             |   50 ++++
 cookbooks/jenkins/providers/job.rb                 |   79 +++++++
 cookbooks/jenkins/providers/node.rb                |   82 +++++++
 cookbooks/jenkins/recipes/default.rb               |  238 ++++++++++++++++++++
 cookbooks/jenkins/recipes/node_jnlp.rb             |   64 ++++++
 cookbooks/jenkins/recipes/node_ssh.rb              |   81 +++++++
 cookbooks/jenkins/recipes/node_windows.rb          |   83 +++++++
 cookbooks/jenkins/recipes/proxy_apache2.rb         |   67 ++++++
 cookbooks/jenkins/recipes/proxy_nginx.rb           |   56 +++++
 cookbooks/jenkins/resources/cli.rb                 |   44 ++++
 cookbooks/jenkins/resources/execute.rb             |   43 ++++
 cookbooks/jenkins/resources/job.rb                 |   35 +++
 cookbooks/jenkins/resources/node.rb                |   54 +++++
 .../jenkins/templates/default/apache_jenkins.erb   |   32 +++
 .../jenkins/templates/default/hudson-slave.xml     |   49 ++++
 .../templates/default/nginx_jenkins.conf.erb       |   33 +++
 .../jenkins/templates/default/port_jenkins.erb     |    2 +
 .../templates/default/sv-hudson-slave-log-run.erb  |    2 +
 .../templates/default/sv-hudson-slave-run.erb      |    6 +
 27 files changed, 1718 insertions(+), 0 deletions(-)
 create mode 100644 cookbooks/jenkins/.gitignore
 create mode 100644 cookbooks/jenkins/README.rdoc
 create mode 100644 cookbooks/jenkins/attributes/default.rb
 create mode 100644 cookbooks/jenkins/files/default/node_info.groovy
 create mode 100644 cookbooks/jenkins/files/ubuntu/jenkins.list
 create mode 100644 cookbooks/jenkins/libraries/manage_node.rb
 create mode 100644 cookbooks/jenkins/metadata.rb
 create mode 100644 cookbooks/jenkins/providers/cli.rb
 create mode 100644 cookbooks/jenkins/providers/execute.rb
 create mode 100644 cookbooks/jenkins/providers/job.rb
 create mode 100644 cookbooks/jenkins/providers/node.rb
 create mode 100644 cookbooks/jenkins/recipes/default.rb
 create mode 100644 cookbooks/jenkins/recipes/node_jnlp.rb
 create mode 100644 cookbooks/jenkins/recipes/node_ssh.rb
 create mode 100644 cookbooks/jenkins/recipes/node_windows.rb
 create mode 100644 cookbooks/jenkins/recipes/proxy_apache2.rb
 create mode 100644 cookbooks/jenkins/recipes/proxy_nginx.rb
 create mode 100644 cookbooks/jenkins/resources/cli.rb
 create mode 100644 cookbooks/jenkins/resources/execute.rb
 create mode 100644 cookbooks/jenkins/resources/job.rb
 create mode 100644 cookbooks/jenkins/resources/node.rb
 create mode 100644 cookbooks/jenkins/templates/default/apache_jenkins.erb
 create mode 100644 cookbooks/jenkins/templates/default/hudson-slave.xml
 create mode 100644 cookbooks/jenkins/templates/default/nginx_jenkins.conf.erb
 create mode 100644 cookbooks/jenkins/templates/default/port_jenkins.erb
 create mode 100644 cookbooks/jenkins/templates/default/sv-hudson-slave-log-run.erb
 create mode 100644 cookbooks/jenkins/templates/default/sv-hudson-slave-run.erb
Cookbook jenkins version 3eec6ac014156ce6f02d3ccb4bb42d3ea49ef1d3 successfully installed
jnewland@zombocom ~/src/cookbooks/dev (master)
→ ruby -v
ruby 1.8.7 (2011-02-18 patchlevel 334) [i686-darwin10.7.0], MBARI 0x6770, Ruby Enterprise Edition 2011.03
```

What ruby version are you running?

**Comment by ringods** [comment]
$ ruby -v
ruby 1.9.2p180 (2011-02-18 revision 30909) [x86_64-linux]

Ruby installed via rvm, running on Ubuntu 10.10 (Desktop machine)

**Comment by jnewland** [comment]
Thanks ringo. I'll check it out under RVM and 1.9.2 later tonight.

**Comment by jnewland** [comment]
attempted to fix in 27960020a13dba9c6db4, issue still remains :/

**Comment by ringods** [comment]
Jesse,

I ran a "gem update" this morning, and retried. It seemed to go well now... Feel free to close this issue.

```
rdesmet@amplidata-desktop-ringo:~/Projects/amplidata/infrastructure/chef-repo$ gem update
Updating installed gems
Updating builder
Fetching: builder-3.0.0.gem (100%)
Successfully installed builder-3.0.0
Updating configuration
Fetching: configuration-1.3.1.gem (100%)
Successfully installed configuration-1.3.1
Updating cucumber
Fetching: gherkin-2.4.1.gem (100%)
Building native extensions.  This could take a while...
Fetching: cucumber-1.0.0.gem (100%)

(::) (::) (::) (::) (::) (::) (::) (::) (::) (::) (::) (::) (::) (::) (::)

Thank you for installing cucumber-1.0.0.
Please be sure to read http://wiki.github.com/cucumber/cucumber/upgrading
for important information about this release. Happy cuking!

(::) (::) (::) (::) (::) (::) (::) (::) (::) (::) (::) (::) (::) (::) (::)

Successfully installed gherkin-2.4.1
Successfully installed cucumber-1.0.0
Updating erubis
Fetching: erubis-2.7.0.gem (100%)
Successfully installed erubis-2.7.0
Updating ffi
Fetching: ffi-1.0.9.gem (100%)
Building native extensions.  This could take a while...
Successfully installed ffi-1.0.9
Updating highline
Fetching: highline-1.6.2.gem (100%)
Successfully installed highline-1.6.2
Updating i18n
Fetching: i18n-0.6.0.gem (100%)
Successfully installed i18n-0.6.0
Updating json
Fetching: json-1.5.3.gem (100%)
Building native extensions.  This could take a while...
Successfully installed json-1.5.3
Updating json_pure
Fetching: json_pure-1.5.3.gem (100%)
Successfully installed json_pure-1.5.3
Updating progressbar
Fetching: progressbar-0.9.1.gem (100%)
Successfully installed progressbar-0.9.1
Updating rake
Fetching: rake-0.9.2.gem (100%)
Successfully installed rake-0.9.2
Updating rest-client
Fetching: rest-client-1.6.3.gem (100%)
Successfully installed rest-client-1.6.3
Updating rspec
Fetching: rspec-core-2.6.4.gem (100%)
Fetching: rspec-expectations-2.6.0.gem (100%)
Fetching: rspec-mocks-2.6.0.gem (100%)
Fetching: rspec-2.6.0.gem (100%)
Successfully installed rspec-core-2.6.4
Successfully installed rspec-expectations-2.6.0
Successfully installed rspec-mocks-2.6.0
Successfully installed rspec-2.6.0
Updating rubygems-update
Fetching: rubygems-update-1.8.5.gem (100%)
Successfully installed rubygems-update-1.8.5
Updating veewee
Fetching: veewee-0.1.26.gem (100%)
Successfully installed veewee-0.1.26
Updating virtualbox
Fetching: virtualbox-0.8.6.gem (100%)
Successfully installed virtualbox-0.8.6
Gems updated: builder, configuration, gherkin, cucumber, erubis, ffi, highline, i18n, json, json_pure, progressbar, rake, rest-client, rspec-core, rspec-expectations, rspec-mocks, rspec, rubygems-update, veewee, virtualbox
Installing ri documentation for builder-3.0.0...
Installing ri documentation for configuration-1.3.1...
Installing ri documentation for gherkin-2.4.1...
Installing ri documentation for cucumber-1.0.0...
Installing ri documentation for erubis-2.7.0...
Installing ri documentation for ffi-1.0.9...
Installing ri documentation for highline-1.6.2...
Installing ri documentation for i18n-0.6.0...
Installing ri documentation for json-1.5.3...
Installing ri documentation for json_pure-1.5.3...
Installing ri documentation for progressbar-0.9.1...
Installing ri documentation for rake-0.9.2...
Installing ri documentation for rest-client-1.6.3...
Installing ri documentation for rspec-core-2.6.4...
Installing ri documentation for rspec-expectations-2.6.0...
Installing ri documentation for rspec-mocks-2.6.0...
Installing ri documentation for rspec-2.6.0...
Installing ri documentation for rubygems-update-1.8.5...
Installing ri documentation for veewee-0.1.26...
Installing ri documentation for virtualbox-0.8.6...
Installing RDoc documentation for builder-3.0.0...
Installing RDoc documentation for configuration-1.3.1...
Installing RDoc documentation for gherkin-2.4.1...
Installing RDoc documentation for cucumber-1.0.0...
Installing RDoc documentation for erubis-2.7.0...
Installing RDoc documentation for ffi-1.0.9...
Installing RDoc documentation for highline-1.6.2...
Installing RDoc documentation for i18n-0.6.0...
Installing RDoc documentation for json-1.5.3...
Installing RDoc documentation for json_pure-1.5.3...
Installing RDoc documentation for progressbar-0.9.1...
Installing RDoc documentation for rake-0.9.2...
Installing RDoc documentation for rest-client-1.6.3...
Installing RDoc documentation for rspec-core-2.6.4...
Installing RDoc documentation for rspec-expectations-2.6.0...
Installing RDoc documentation for rspec-mocks-2.6.0...
Installing RDoc documentation for rspec-2.6.0...
Installing RDoc documentation for rubygems-update-1.8.5...
Installing RDoc documentation for veewee-0.1.26...
Installing RDoc documentation for virtualbox-0.8.6...
rdesmet@amplidata-desktop-ringo:~/Projects/amplidata/infrastructure/chef-repo$ knife cookbook github install fnichol/chef-jenkins -V
Installing jenkins from git://github.com/fnichol/chef-jenkins.git to /home/rdesmet/Projects/amplidata/infrastructure/chef-repo/.chef/../cookbooks
Checking out the master branch.
Pristine copy branch (chef-vendor-jenkins) exists, switching to it.
Removing pre-existing version.
1 files updated, committing changes
Creating tag cookbook-site-imported-jenkins-3eec6ac014156ce6f02d3ccb4bb42d3ea49ef1d3
Checking out the master branch.
Updating f5a292e..03cba0a
Fast-forward
cookbooks/jenkins/.gitignore                       |    2 +
cookbooks/jenkins/README.rdoc                      |  188 +++++++++++++++
cookbooks/jenkins/attributes/default.rb            |  122 ++++++++++
cookbooks/jenkins/files/default/node_info.groovy   |   86 +++++++
cookbooks/jenkins/files/ubuntu/jenkins.list        |    1 +
cookbooks/jenkins/libraries/manage_node.rb         |  158 +++++++++++++
cookbooks/jenkins/metadata.rb                      |    7 +
cookbooks/jenkins/providers/cli.rb                 |   54 +++++
cookbooks/jenkins/providers/execute.rb             |   50 ++++
cookbooks/jenkins/providers/job.rb                 |   79 +++++++
cookbooks/jenkins/providers/node.rb                |   82 +++++++
cookbooks/jenkins/recipes/default.rb               |  238 ++++++++++++++++++++
cookbooks/jenkins/recipes/node_jnlp.rb             |   64 ++++++
cookbooks/jenkins/recipes/node_ssh.rb              |   81 +++++++
cookbooks/jenkins/recipes/node_windows.rb          |   83 +++++++
cookbooks/jenkins/recipes/proxy_apache2.rb         |   67 ++++++
cookbooks/jenkins/recipes/proxy_nginx.rb           |   56 +++++
cookbooks/jenkins/resources/cli.rb                 |   44 ++++
cookbooks/jenkins/resources/execute.rb             |   43 ++++
cookbooks/jenkins/resources/job.rb                 |   35 +++
cookbooks/jenkins/resources/node.rb                |   54 +++++
.../jenkins/templates/default/apache_jenkins.erb   |   32 +++
.../jenkins/templates/default/hudson-slave.xml     |   49 ++++
.../templates/default/nginx_jenkins.conf.erb       |   33 +++
.../jenkins/templates/default/port_jenkins.erb     |    2 +
.../templates/default/sv-hudson-slave-log-run.erb  |    2 +
.../templates/default/sv-hudson-slave-run.erb      |    6 +
27 files changed, 1718 insertions(+), 0 deletions(-)
create mode 100644 cookbooks/jenkins/.gitignore
create mode 100644 cookbooks/jenkins/README.rdoc
create mode 100644 cookbooks/jenkins/attributes/default.rb
create mode 100644 cookbooks/jenkins/files/default/node_info.groovy
create mode 100644 cookbooks/jenkins/files/ubuntu/jenkins.list
create mode 100644 cookbooks/jenkins/libraries/manage_node.rb
create mode 100644 cookbooks/jenkins/metadata.rb
create mode 100644 cookbooks/jenkins/providers/cli.rb
create mode 100644 cookbooks/jenkins/providers/execute.rb
create mode 100644 cookbooks/jenkins/providers/job.rb
create mode 100644 cookbooks/jenkins/providers/node.rb
create mode 100644 cookbooks/jenkins/recipes/default.rb
create mode 100644 cookbooks/jenkins/recipes/node_jnlp.rb
create mode 100644 cookbooks/jenkins/recipes/node_ssh.rb
create mode 100644 cookbooks/jenkins/recipes/node_windows.rb
create mode 100644 cookbooks/jenkins/recipes/proxy_apache2.rb
create mode 100644 cookbooks/jenkins/recipes/proxy_nginx.rb
create mode 100644 cookbooks/jenkins/resources/cli.rb
create mode 100644 cookbooks/jenkins/resources/execute.rb
create mode 100644 cookbooks/jenkins/resources/job.rb
create mode 100644 cookbooks/jenkins/resources/node.rb
create mode 100644 cookbooks/jenkins/templates/default/apache_jenkins.erb
create mode 100644 cookbooks/jenkins/templates/default/hudson-slave.xml
create mode 100644 cookbooks/jenkins/templates/default/nginx_jenkins.conf.erb
create mode 100644 cookbooks/jenkins/templates/default/port_jenkins.erb
create mode 100644 cookbooks/jenkins/templates/default/sv-hudson-slave-log-run.erb
create mode 100644 cookbooks/jenkins/templates/default/sv-hudson-slave-run.erb
Cookbook jenkins version 3eec6ac014156ce6f02d3ccb4bb42d3ea49ef1d3 successfully installed
rdesmet@amplidata-desktop-ringo:~/Projects/amplidata/infrastructure/chef-repo$ 
```
